pandas/tests/series/methods/test_quantile.py

```diff
@@ -245,3 +245,10 @@ def test_quantile_dtype_size(self, any_int_ea_dtype):
         result = ser.quantile([0.1, 0.5])
         expected = Series([1, 1], dtype=any_int_ea_dtype, index=[0.1, 0.5])
         tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("typ", ["datetime64", "timedelta64"])
+def test_quantile_empty_datetimelike(typ, unit):
```

Comment:
Maybe it would be good to also add a dataframe version of this test, such that there is an actual dtype to verify (to assert it has preserved the unit)